teps/0052-nested-triggers.md

```diff
@@ -0,0 +1,202 @@
+downstream trigger must be excluded from processing in the default HTTP endpoint. One of the goals
+outlined above is that Trigger authors are enabled to focus on the logic specific to their trigger.
+If an interceptor chain is removed from the processing of an authored trigger then it cannot be
+exposed for direct requests otherwise requests could bypass the desired filtering.
+
+#### Example
+
+The following is a set of examples of how you could invoke nested triggers within an event listener.
+Note that this example is tied to the example implementation discussed above, but is intended to demonstrate
+some of the power of making this feature available.
+First, a top level trigger invokes nested triggers:
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: top-level-trigger
+  labels:
+    tekton.dev/sink: exposed
+spec:
+  interceptors:
+    #could be github interceptor as well, just using cel as example
+    - cel:
+        filter: >-
+          header.match('header-filter', 'check-this-header-filter')
+        - key: example-overlay
+          expression: body.example.value
+    - webhook:
+        objectRef:
+          kind: Service
+          name: example-webhook-service
+          apiVersion: v1
+          namespace: tekton-pipelines
+    #webhook doesn't explicitly set extensions
+    - cel:
+        overlays:
+        - key: example-webhook
+          expression: body.extensions.webhook-value
+  triggers:
+  - ref: nested-trigger-1
+  - ref: nested-trigger-2
+```
+
+Then, the downstream triggers that are invoked do not need the common interceptors and can
+specify their own triggerbindings and templates.
+
+```yaml
+#nested-trigger-1
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: nested-trigger-1
+  labels:
+    #trigger is not available for external processing
+    tekton.dev/sink: hidden
+spec:
+  interceptors:
+  - cel:
+      filter: >-
+        extensions.example-webhook == "filtered-value-1"
+  bindings:
+  - ref: nested-binding-1
+  - ref: nested-binding-2
+  template:
+    ref: nested-triggertemplate-1
+```
+
+```yaml
+#nested-trigger-2
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: nested-trigger-2
+  labels:
+    #trigger is not available for external processing
+    tekton.dev/sink: hidden
+spec:
+  interceptors:
+  - cel:
+      filter: >-
+        extensions.example-webhook == "filtered-value-2"
+  triggers:
+  #note: not shown here
+  - ref: nested-trigger-3
+  bindings:
+  - ref: nested-binding-3
+  - ref: nested-binding-4
+  template:
+    ref: nested-triggertemplate-2
+```
+
+Notice how simple it is for the Trigger author to write the logic around their trigger
+when the common logic is implemented at the top level. Trigger authors can also manage
+their own set of downstream triggers as you can specify additional triggers alongside
+a binding and template once the interceptor chain completes.
```

Comment:
seeing these points i am thinking like it would be great if you add `Advantages` section and add these points over there

WDYT?

Reply:
I think it's probably fine to merge this as is as proposed and then come back and add in some alternatives and flesh out the design.